services/web-app/lib/github.js

```diff
@@ -647,22 +647,26 @@ export const getLibraryData = async (params = {}) => {
    */
   let response = {}
 
+  const { host, org, repo, path: libPath, ref } = libraryParams
+
   try {
-    response = await getResponse(libraryParams.host, 'GET /repos/{owner}/{repo}/contents/{path}', {
-      owner: libraryParams.org,
-      repo: libraryParams.repo,
-      path: removeLeadingSlash(`${libraryParams.path}/carbon.yml`),
-      ref: libraryParams.ref
+    response = await getResponse(host, 'GET /repos/{owner}/{repo}/contents/{path}', {
+      owner: org,
+      repo,
+      path: removeLeadingSlash(`${libPath}/carbon.yml`),
+      ref
     })
   } catch (err) {
     return null
   }
 
   let content
+  const dereferenceKey = `library=${host}/${org}/${repo}&path=${libPath}&ref=${ref}-dereferenced`
```

Comment:
`file-cache.js` has a private `slugifyRequest()` that generates cache keys. Would it be good to use that function to create the dereference key?

Reply:
Maybe not. I see those keys look like:

`github.com GET /repos/francinelucca/library-testing/contents/carbon.yml ref=main`

Instead of:

`library=github.com/francinelucca/library-testing&path=/&ref=main-dereferenced`

Maybe match that format with:

`github.com GET DEREFERENCED /repos/francinelucca/library-testing/contents/carbon.yml ref=main`

?

Not a big deal. I can't think of a scenario when your key wouldn't be unique. If you want to reuse `slugifyRequest()`, you could just pass in if it's a dereferenced request.

Reply:
using `slugifyRequest` to generate key now!